Dockerfile

```diff
@@ -0,0 +1,140 @@
+# All build arguments
+
+# TARGETPLATFORM - automatic assignment via `docker build --platform xyz`
+# RUST_PROFILE - defaults to release
+
+# Build babylon-node binary
+FROM debian:11-slim AS binary-build-stage
+LABEL org.opencontainers.image.authors="devops@radixdlt.com"
+LABEL org.opencontainers.image.description="Java + Debian 11 (OpenJDK)"
+
+ENV DEBIAN_FRONTEND noninteractive
+
+CMD ["/bin/bash"]
+
+RUN  \
+  apt-get update && apt-get install -y --no-install-recommends \
+  docker.io=20.10.5+dfsg1-1+deb11u2 \
+  libssl-dev=1.1.1n-0+deb11u4 \
+  pkg-config=0.29.2-1 \
+  unzip=6.0-26+deb11u1 \
+  wget=1.21-1+deb11u1 \
+  software-properties-common=0.96.20.2-2.1 
+RUN add-apt-repository -y ppa:openjdk-r/ppa && \
+  apt-get install -y --no-install-recommends \
+  openjdk-17-jdk=17.0.6+10-1~deb11u1
+
+#install Gradle
+RUN wget -q https://services.gradle.org/distributions/gradle-7.2-bin.zip \
+  && unzip gradle-7.2-bin.zip -d /opt \
+  && rm gradle-7.2-bin.zip
+
+# Set Gradle in the environment variables
+ENV GRADLE_HOME=/opt/gradle-7.2
+ENV PATH=/opt/gradle-7.2/bin:$PATH
+ENV JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
+
+COPY . /radixdlt
+WORKDIR /radixdlt
+USER root
+RUN SKIP_NATIVE_RUST_BUILD=TRUE gradle clean build -x test -Pci=true -PrustBinaryBuildType=release
+
+USER nobody
+
+# Exporting babylon-node binary
+FROM scratch AS binary-container
+COPY --from=binary-build-stage /radixdlt/core/build/distributions /
+
+# Building core-rust library
+FROM lukemathwalker/cargo-chef:0.1.52-rust-1.68.0 AS chef
+LABEL org.opencontainers.image.authors="devops@radixdlt.com"
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    clang=1:11.0-51+nmu5 \
+    libmpc-dev=1.2.0-1 \
+    libmpfr-dev=4.1.0-3 \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
Can you explain why we do this line?

Reply:
The apt-get clean and rm -rf /var/lib/apt/lists/* reduce the image size after installation. This is done automatically on ubuntu based images. On debian we have to do this ourselves.

The installed packages seem to be dependencies to build our specific rust target.